src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -7,18 +7,130 @@
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private static double threshold;
+    private int size;
+    private int capacity;
+    Node<K, V>[] table;
+
+    public MyHashMap() {
+        size = 0;
+        capacity = DEFAULT_CAPACITY;
+        table = new Node[capacity];
+        threshold = capacity * LOAD_FACTOR;
+    }
+
     @Override
     public void put(K key, V value) {
-
+        if (value == null) {
+            return;
+        }
+        if (key == null) {
+            putWithNullKey(value);
+        } else {
+            int hash = hash(key.hashCode());
+            int index = indexFor(hash);
+            if (table[index] == null) {
+                addEntry(key, value, index);
+                ++size;
+            } else {
+                Node<K, V> nodeToCheck = table[index];
+                while (nodeToCheck != null) {
+                    if (key.equals(nodeToCheck.key)) {
+                        nodeToCheck.value = value;
+                        return;
+                    }
+                    if (value.equals(nodeToCheck.value)) {
+                        return;
+                    }
+                    nodeToCheck = nodeToCheck.next;
+                }
+                addEntry(key, value, index);
+                ++size;
+            }
+        }
+        if (size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (key == null) {
+            return table[0] == null ? null : table[0].value;
+        }
+        int hash = hash(key.hashCode());
+        int index = indexFor(hash);
+        if (index < 0 || index >= table.length) {
+            return null;
+        }
+        if (table[index] == null) {
+            return  null;
+        }
+        Node<K, V> nodeToCheck = table[index];
+        while (nodeToCheck != null) {
+            if (key.equals(nodeToCheck.key)) {
+                return nodeToCheck.value;
+            }
+            nodeToCheck = nodeToCheck.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        capacity *= 2;
+        Node<K, V>[] oldTable = table;
+        table = new Node[capacity];
+        for (int i = 0; i < oldTable.length; i++) {
+            Node<K, V> node = oldTable[i];
+            while (node != null) {
+                int hash = hash(node.key.hashCode());
+                int index = indexFor(hash);
+                table[index] = new Node<K, V>(node.key, node.value, null);
+                node = node.next;
+            }
+        }
+        threshold = capacity * LOAD_FACTOR;
+    }
+
+    private void putWithNullKey(V value) {
+        if (table[0] == null) {
```

Comment:
do we need this checking here?

Reply:
I think, we do, as we have to know, whether we create new Node or only overwrite the value of existing one